bundles/framework/announcements/tool/AnnouncementsTool.js

```diff
@@ -0,0 +1,367 @@
+                        if (toolPluginMapfullConf.config.announcements[i].id == '' + id) {
+                            return true;
+                        }
+                    }
+                } else {
+                    return false;
+                }
+            } else {
+                return false;
+            }
+        },
+        /**
+         * @private @method _getToolPluginMapfullConf
+         * Get map view cofiguration (from mapfull) for this tool
+         * @return {Object / null} config or null if not found
+         */
+        _getToolPluginMapfullConf: function () {
+            var me = this;
+            var isConfig = !!((me.data && me.data.configuration));
+            var isPlugins = !!((isConfig && me.data.configuration.mapfull &&
+            me.data.configuration.mapfull.conf && me.data.configuration.mapfull.conf.plugins));
+            var toolPlugin = null;
+            if (isPlugins) {
+                var plugins = me.data.configuration.mapfull.conf.plugins;
+                for (var i = 0; i < plugins.length; i++) {
+                    var plugin = plugins[i];
+                    if (plugin.id === me.getTool().id) {
+                        toolPlugin = plugin;
+                        break;
+                    }
+                }
+            }
+            return toolPlugin;
+        },
+
+        fetchAnnouncements: function () {
+            jQuery.ajax({
+                type: 'GET',
+                dataType: 'json',
+                url: Oskari.urls.getRoute('Announcements'),
+                success: (pResp) => {
+                    this.announcements = pResp.data;
+                },
+                error: function (jqXHR, textStatus) {
+                }
+            });
+        },
+
+        _getAnnouncementsSelection: function () {
+            var me = this,
+                announcementsSelection = {};
+            var pluginValues = me.getPlugin().getSelectedAnnouncements();
+            if (pluginValues.announcements) {
+                announcementsSelection.announcements = pluginValues.announcements;
+            }
+            return announcementsSelection;
+        },
+
+        /**
+        * Get values.
+        * @method getValues
+        * @public
+        *
+        * @returns {Object} tool value object
+        */
+        getValues: function () {
+            var me = this;
+
+            if (me.state.enabled) {
+                var pluginConfig = { id: this.getTool().id, config: this.getPlugin().getConfig() };
+                var announcementsSelection = me._getAnnouncementsSelection();
+
+                if (announcementsSelection && !jQuery.isEmptyObject(announcementsSelection)) {
+                    pluginConfig.config.announcements = announcementsSelection.announcements;
+                }
+                return {
+                    configuration: {
+                        mapfull: {
```

Comment:
Reconsider having this announcements bundle as part of mapfull plugins. A weird dependency is created for mapfull here if it's part of these plugins.

Reply:
In my opinion the plugin should be part of the announcements bundle and not part of the "mapmodule" plugins.

Reply:
Reason being that if it's part of mapmodule plugins _everyone_ will need to have that code loaded to the browser even if the Oskari instance isn't using the announcements functionality. So it will increase the size of the js-file end-users need to load to get the application up and running even when the functionality isn't used.